File: Examples/VisualBasic/PerformingandManagingOCR/PerformOCROnImage.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.IO

Imports Aspose.OCR

Namespace Aspose.OCR.Examples.VisualBasic.PerformingandManagingOCR
    Public Class PerformOCROnImage
        Public Shared Sub Run()
            ' ExStart:PerformOCROnImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_OCR()

            ' Source file: the file on which OCR will be performed.
            Dim imageFile As String = dataDir & "Sampleocr.bmp"

            Console.WriteLine("Performing OCR on " & imageFile & "....")

            ' Initialize OcrEngine.
            Dim ocr As New OcrEngine()
            ' Set the image.
            ocr.Image = ImageStream.FromFile(imageFile)

            Try
                ' Process the whole image
                If ocr.Process() Then
                    ' Get the complete recognized text found from the image
                    Console.WriteLine("Text recognized: " & ocr.Text.ToString())
                    File.WriteAllText(dataDir & "Output.txt", ocr.Text.ToString())
                End If
            Catch ex As Exception
                Console.WriteLine("Exception: " & ex.ToString())
            End Try
            ' ExEnd:PerformOCROnImage
        End Sub
    End Class
End Namespace

File: Examples/CSharp/PerformingandManagingOCR/PerformOCROnImage.cs
using System;
using System.IO;
using Aspose.OCR;

namespace Aspose.OCR.Examples.CSharp.PerformingandManagingOCR
{
    public class PerformOCROnImage
    {
        public static void Run()
        {
            // ExStart:PerformOCROnImage   
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_OCR();

            // Initialize an instance of OcrEngine
            OcrEngine ocrEngine = new OcrEngine();

            // Set the Image property by loading the image from file path location or an instance of MemoryStream 
            ocrEngine.Image = ImageStream.FromFile(dataDir + "Sampleocr.bmp");

            // Process the image
            if (ocrEngine.Process())
            {
                // Display the recognized text
                Console.WriteLine(ocrEngine.Text);
            }
            // ExEnd:PerformOCROnImage   
        }
    }
}
